package NReqBundleIteratorProto;


message TConstraintInfo {
    optional uint32 Index = 1;
    optional uint32 Type = 2;
    repeated uint32 BlockIndices = 3;
}

message TConstraints {
    repeated TConstraintInfo Constraints = 1;
}

message TConstraintCheckerOptions {
    optional bool IgnoreZeroBreak = 1;
    optional bool CheckQuotedConstraint = 2;
}

message TConstraintChecker {
    repeated TConstraints ConstraintsByFirstBlock = 1;

    optional TConstraintCheckerOptions Options = 2;

    optional uint32 ConstraintCount = 3;

    optional bool NeedSentenceLengths = 4;

    repeated uint32 MultitokenBlockIndex = 5;

    repeated bool IsAnyBlock = 6;

    repeated bool IsOriginalRequestBlock = 7;

    repeated uint32 MultitokenBlockWordCount = 8;
}
